airflow/operators/slack_operator.py

```diff
@@ -116,3 +118,70 @@ def construct_api_call_params(self):
             'icon_url': self.icon_url,
             'attachments': json.dumps(self.attachments),
         }
+
+class SlackWebHookHook(BaseHook):
+    """Slack WebHook hook. Connects to a slack webhook URL using a Connection.
+
+    The Slack WebHook URL should be defined in the 'extra' field of the connection as defined
+    in the Airflow admin interface.
+    """
+    def __init__(self, conn_id):
+        self.conn_id = conn_id
+        self.webhook_url = self.get_connection(conn_id).extra
```

Comment:
You might want to be user friendly here as 'extra' is normally not required so some logging in case it isn't available could help.